src/Plugin/RulesExpression/ActionSet.php

```diff
@@ -77,7 +89,7 @@ public function addExpressionObject(ExpressionInterface $expression) {
     if (!$expression instanceof ActionExpressionInterface) {
       throw new InvalidExpressionException();
     }
-    $this->actions[] = $expression;
+    $this->actions[$this->uuidService->generate()] = $expression;
```

Comment:
I don't think this belongs on the expression manager. It is just one call to generate a UUID, I would not couple it to the manger since it is none of the managers business where I get my UUIds from.

Implemented the lookup method as getExpression().